pkglistgen.py

```diff
@@ -1146,6 +1146,7 @@ def do_update_and_solve(self, subcmd, opts):
             # TODO Continue supporting #1297, but should be abstracted.
             main_repo = 'ports'
             opts.project += ':Ports'
+            target_project += ':Ports'
```

Comment:
When `target_project` is then passed to `project_list_family_prior()` the following invalid output is returned since `:Ports` isn't a base product name.

```python
['openSUSE:Leap:15.0:Ports', 'openSUSE:Leap:15.0:Rings', 'openSUSE:Leap:15.0:SLE-workarounds', 'openSUSE:Leap:15.0:Staging', 'openSUSE:Leap:15.0:ToTest']
```

Whereas without suffix:

```python
['openSUSE:Leap:15.0', 'openSUSE:Leap:42.3', 'openSUSE:Leap:42.2', 'openSUSE:Leap:42.1']
```

Presumably you want this so that the drop list functions and retrieves `:Ports` solv files for prior Leap releases? If so the drop list generation will need to be worked on a bit as the download urls come from the config for the product release and do not contain the `:Ports` paths.

As such this should be resolved differently as `target_project` is intended to be the "main release product" just like it is `openSUSE:Leap:15.0` even for `openSUSE:Leap:15.0:Staging:A`.

Basically the lines in `solv_cache_update()` where it gets the `baseurl` need to figure out the `:Ports` path. Otherwise I imagine it works.